apps/condo/domains/marketplace/schema/MarketItemPrice.test.js

```diff
@@ -0,0 +1,345 @@
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await updateTestMarketItemPrice(client, objCreated.id)
+                })
+            })
+
+            test('can\'t delete', async () => {
+                await expectToThrowAccessDeniedErrorToObj(async () => {
+                    await MarketItemPrice.delete(client, 'id')
+                })
+            })
+
+            test('can read MarketItemPrice of his organization', async () => {
+                const [obj] = await createTestMarketItemPrice(admin, marketItem)
+
+                const objs = await MarketItemPrice.getAll(client, {}, { sortBy: ['updatedAt_DESC'] })
+                expect(objs.length).toBeGreaterThanOrEqual(1)
+                expect(objs).toEqual(expect.arrayContaining([
+                    expect.objectContaining({
+                        id: obj.id,
+                    }),
+                ]))
+            })
+            // TODO (DOMA-7503) test('Resident can read items according to PriceScope settings')
+        })
+
+    })
+
+    describe('Hooks validation tests', () => {
+        let marketItem
+        beforeAll(async () => {
+            [marketItem] = await createTestMarketItem(admin, marketCategory, organization)
+        })
+        test('Validate item price', async () => {
+            await expectToThrowGQLError(
+                async () => {
+                    await createTestMarketItemPrice(admin, marketItem, {
+                        price: [{ ...validPriceFieldValue[0], price: '-1' }],
+                    })
+                },
+                {
+                    code: 'BAD_USER_INPUT',
+                    type: 'INVALID_PRICE',
+                },
+                'obj'
+            )
+        })
+        test('Validate salesTaxPercent', async () => {
+            await expectToThrowGQLError(
+                async () => {
+                    await createTestMarketItemPrice(admin, marketItem, {
+                        price: [{ ...validPriceFieldValue[0], salesTaxPercent: '-1' }],
+                    })
+                },
+                {
+                    code: 'BAD_USER_INPUT',
+                    type: 'INVALID_SALES_TAX_PERCENT',
+                },
+                'obj'
+            )
+        })
+    })
+
+    describe('Price field validation tests', () => {
+        test('positive validation test', async () => {
+            expect(validatePriceField(validPriceFieldValue)).toEqual(true)
+        })
+
+        describe('Negative tests', () => {
+            const cases = [
+                [{ ...validPriceFieldValue, type: 'invalid' }],
+                [{ ...validPriceFieldValue, group: [] }],
+                [{ ...validPriceFieldValue, name: [] }],
+                [{ ...validPriceFieldValue, price: 'invalid' }],
+                [{ ...validPriceFieldValue, isMin: 'invalid' }],
+                [{ ...validPriceFieldValue, vatPercent: '100' }],
```

Comment:
Should the sales tax percent field be tested here?

Reply:
yes, this is not a validation check in hooks, but an `ajv` validation check. just need to add hook tests next to it

Reply:
above there is a separate block for this `describe('Hooks validation tests'`